examples/material-ui/src/example/core.cljs

```diff
@@ -7,42 +7,28 @@
 (def mui-theme-provider (r/adapt-react-class mui/MuiThemeProvider))
 (def menu-item (r/adapt-react-class mui/MenuItem))
 
-(def ^:private input-component
-  (r/reactify-component
-    (fn [props]
-      [:input (-> props
-                  (assoc :ref (:inputRef props))
-                  (dissoc :inputRef))])))
-
-(def ^:private textarea-component
-  (r/reactify-component
-    (fn [props]
-      [:textarea (-> props
-                     (assoc :ref (:inputRef props))
-                     (dissoc :inputRef))])))
-
-;; To fix cursor jumping when controlled input value is changed,
-;; use wrapper input element created by Reagent instead of
-;; letting Material-UI to create input element directly using React.
-;; Create-element + convert-props-value is the same as what adapt-react-class does.
-(defn text-field [props & children]
-  (let [props (-> props
-                  (assoc-in [:InputProps :inputComponent] (cond
-                                                            (and (:multiline props) (:rows props) (not (:maxRows props)))
-                                                            textarea-component
-
-                                                            ;; FIXME: Autosize multiline field is broken.
-                                                            (:multiline props)
-                                                            nil
-
-                                                            ;; Select doesn't require cursor fix so default can be used.
-                                                            (not (:select props))
-                                                            nil
-
-                                                            :else
-                                                            input-component))
-                  rtpl/convert-prop-value)]
-    (apply r/create-element mui/TextField props (map r/as-element children))))
+(defn adapt-input-component [component]
+  (fn [props & _]
+    (r/create-class
+      {:getInitialState (fn [] #js {:value (:value props)})
```

Comment:
AFAIK state doesn't have to be a map so you could just use `(:value props)` here